File: apps/flutter/lib/features/substitutions/substitution.dart
import 'package:class_mate/api/types.dart';

enum SubstitutionType {
  freisetzung,
  vertretung,
  betreuung,
  entfall,
}

final typeMap = {
  SubstitutionsGetOutputTypeEnum.betreuung: SubstitutionType.betreuung,
  SubstitutionsGetOutputTypeEnum.entfall: SubstitutionType.entfall,
  SubstitutionsGetOutputTypeEnum.freisetzung: SubstitutionType.freisetzung,
  SubstitutionsGetOutputTypeEnum.vertretung: SubstitutionType.vertretung,
};

final typeStringMap = {
  SubstitutionType.betreuung: "Betreuung",
  SubstitutionType.entfall: "Entfall",
  SubstitutionType.freisetzung: "Freisetzung",
  SubstitutionType.vertretung: "Vertretung",
};

class Substitution {
  SubstitutionType type;

  Substitution({
    required this.type,
  });
}

File: packages/lib/src/substitutions/substitution.ts
export interface Substitution {
  date: Date;
  type: SubstitutionType | null;
}

export const SUBSTITUTION_TYPES = [
  "FREISETZUNG",
  "VERTRETUNG",
  "BETREUUNG",
  "ENTFALL",
  "TROTZ_ABSENZ",
] as const;

export type SubstitutionType = (typeof SUBSTITUTION_TYPES)[number];
